### PR TITLE
MacOS fixes

### PR DESCRIPTION
**Pull request by mattisz**
Goals of this PR:

1. Build for apple silicon with code-signing and notarization
2. Update some packages
3. Minimize changes to build process

What's in this PR:

1. Updates electron-builder to 25.1.3 to leverage builtin notarization step for macOS.
2. Removes `com.apple.security.cs.allow-unsigned-executable-memory` entitlement as we are using electron 12+ - per [electron/notarize](https://github.com/electron/notarize?tab=readme-ov-file#prerequisites).
3. Adds the `com.apple.security.cs.allow-jit` entitlement to allow arm64 builds as we are using electron 20+ - per [electron-builder mac entitlements](https://www.electron.build/mac.html#entitlements).
4. Removes legacy `internals/scripts/Notarize.js` and associated afterSign section from [package.json](/tom-james-watson/breaktimer-app/blob/master/package.json) in favor of electron-builder's builtin notarization.
5. Updates [package-lock.json](/tom-james-watson/breaktimer-app/blob/master/package-lock.json) per `npm audit fix`.
6. Updates [DEVELOPMENT.md](/tom-james-watson/breaktimer-app/blob/master/DEVELOPMENT.md) to include environment variable info for MacOS signing/notarization.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -49,6 +49,12 @@ To package apps for the local platform:
 npm run package
 ```
 
+:bulb: When packaging for MacOS make sure to set your signing/notirization env variables per [the electron-builder docs](https://www.electron.build/mac#notarize):
+
+```bash
+APPLE_ID='YOUR_APPLE_ID' APPLE_APP_SPECIFIC_PASSWORD='YOUR_APP_PASSWORD' APPLE_TEAM_ID='YOUR_TEAM_ID' npm run package
```
Comment (tom-james-watson) [inline DEVELOPMENT.md:55]: They recommend using APPLE_API_KEY, APPLE_API_KEY_ID and APPLE_API_ISSUER. We should update the docs to use that instead 🙏

Comment (tom-james-watson) [reply DEVELOPMENT.md:55]: Notarizing was also not working until I added `notarize: true` to the `mac` block in `package.json`.

Comment (tom-james-watson) [reply DEVELOPMENT.md:55]: I'm running an arm mac and this makes a pure arm64 build, which means this wouldn't work on older intel macs. I think we need to build a universal dmg.

Comment (tom-james-watson) [reply DEVELOPMENT.md:55]: If you replace

```
      "target": [
        "dmg"
      ],
```

with

```
      "target": [
        {
          "target": "dmg",
          "arch": [
            "universal"
          ]
        }
      ],
```

then it builds a universal dmg.

**Comment by tom-james-watson** [comment]
Thanks for this - this is waaaay overdue and was holding me back from bothering to make updates to the app 🙏.

If you can make those changes then we can sort out making a new release.

File: DEVELOPMENT.md (hunk 1 above)
Comment (tom-james-watson) [reply DEVELOPMENT.md:55]: Also, can you add a note for something like

```
You can obtain an API key from [App Store Connect](https://appstoreconnect.apple.com/access/integrations/api).
Create a **Team Key** (not an _Individual Key_) with **App Manager** access.
```

**Comment by mattisz** [comment]
@tom-james-watson Always happy to help support a project I use every day.

I got those changes sorted out, built a universal dmg, and everything is working as expected on my mac. However, I do not have an Intel Mac to test on.

Let me know if there is anything else I can do to help.

**Comment by tom-james-watson** [comment]
Just need to rebase on master and it should be good to go!

**Comment by mattisz** [comment]
@tom-james-watson Rebased

**Review by tom-james-watson — APPROVED** [review]
Thanks! 🚀

**Pull request merged**